src-tauri/src/depotdownloader.rs

```diff
@@ -0,0 +1,115 @@
+use std::fs::File;
+use std::io::ErrorKind::AlreadyExists;
+use std::{fs, io};
+use std::{io::Write, path::Path};
+
+use crate::get_os;
+use reqwest;
+use sha256;
+
+pub static DEPOTDOWNLOADER_VERSION: &str = "2.7.3";
+
+pub fn calc_checksum(path: &Path) -> io::Result<String> {
+    let bytes = fs::read(path)?;
+    let hash = sha256::digest(&bytes);
+    Ok(hash)
+}
+
+/**
+See: [`test_get_depotdownloader_url()`]
+*/
+pub fn get_depotdownloader_url() -> String {
+    let arch = match std::env::consts::ARCH {
+        "x86_64" => "x64",
+        "aarch64" => "arm64",
+        "arm" => "arm",
+        _ => "x86_64",
+    };
+
+    format!("https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_{}/DepotDownloader-{}-{}.zip", DEPOTDOWNLOADER_VERSION, get_os(), arch)
+}
+
+/// Downloads a file. The file will be saved to the [`filename`] provided.
+pub async fn download_file(url: &str, filename: &Path) -> io::Result<()> {
+    if filename.exists() {
+        println!("DEBUG: Not downloading. File already exists.");
+        return Err(io::Error::from(AlreadyExists));
+    }
+
+    let mut file = File::create(filename)?;
+    let response = reqwest::get(url)
+        .await
+        .expect("Failed to contact internet.");
```

Comment:
you could probably skip the next part where it tries to get the data from the request if it failed. 

A second error message from the same problem could get confused as 2 problems.

Same idea with if the content errors. should not try to write to the file if either of these 2 error